src/language-js/utils.js

```diff
@@ -342,8 +342,28 @@ function isGetterOrSetter(node) {
  * @param {Node} nodeB
  * @returns {boolean}
  */
-function sameLocStart(nodeA, nodeB, options) {
-  return options.locStart(nodeA) === options.locStart(nodeB);
+function sameLocStart(nodeA, nodeB, { locStart }) {
+  return locStart(nodeA) === locStart(nodeB);
+}
+
+/**
+ * @param {Node} nodeA
+ * @param {Node} nodeB
+ * @returns {boolean}
+ */
+function sameLocEnd(nodeA, nodeB, { locEnd }) {
+  return locEnd(nodeA) === locEnd(nodeB);
+}
+
+/**
+ * @param {Node} nodeA
+ * @param {Node} nodeB
+ * @returns {boolean}
+ */
+function isSameNode(nodeA, nodeB, options) {
```

Comment:
`isSameLoc` is a better name. Different nodes can have the same location. E.g.
```json
      {
        "type": "ExpressionStatement",
        "start": 0,
        "end": 3,
        "loc": {
          "start": { "line": 1, "column": 0 },
          "end": { "line": 1, "column": 3 }
        },
        "range": [0, 3],
        "expression": {
          "type": "Identifier",
          "start": 0,
          "end": 3,
          "loc": {
            "start": { "line": 1, "column": 0 },
            "end": { "line": 1, "column": 3 },
            "identifierName": "fff"
          },
          "range": [0, 3],
          "name": "fff"
        }
      }
```

Reply:
Renamed as `hasSameLoc` https://github.com/prettier/prettier/pull/9435/commits/0d515b93a89b51b1b70bc4e31a663bc0caf75156